### DISCUSSION TITLE
Updating spatial grid for spatial interaction model

### DISCUSSION THREAD

**Discussion by Robinlovelace** [nptscot/npt#485] [Ideas]
Currently we're using this code to get a fixed grid of points 500 m apart in x and y directions:

https://github.com/nptscot/npt/blob/f420b5f9872cb8de0a867a5dfcb2cdd81682e2bf/_targets.R#L748-L752

One idea in https://github.com/nptscot/npt/blob/main/report-utility-methods.md#next-steps is:

> Revisit the grid: should a more approach e.g. involving spatial clustering of points be used?

I know @Hussein-Mahfouz has looked at spatial cluster with DBSCAN. Could this be useful here?

Cc @joeytalbot and @wangzhao0217 as discussed today.

Not an issue just a Phase 3 idea so far hence putting in discussion not issue.

**Comment by Hussein-Mahfouz** [comment]
Is this related to the work on sampling subpoints for odjitter? Related atumie issue https://github.com/ITSLeeds/atumie/issues/975
Or is this a different problem?

**Reply by Robinlovelace** [reply]
Different but related.

Instead of using all possible points (POIs from OS data primarily), we're using a regular grid of points as the subpoints, with the weight related to the characteristics (currently just the number) of trip attractors in each cell. The idea is to not use a regular 500 m grid but a grid that becomes smaller towards places with high population/activity density and larger in more remote/rural areas.

**Reply by Hussein-Mahfouz** [reply]
Short answer: I'm not sure - we'd have to try to find out. Longer answer:

DBSCAN takes two parameters: epsilon (search radius) and minpts (minimum number of points required to form a cluster). These parameters are set globally, so you can't optimise for areas of different point density simultaneously. Take two scenarios

1) Radius is small (better for areas with high POI density)
    - In high density areas, it will cluster nearby points
    - In low density areas, the radius may be too small to cluster points

2) Radius is big (better for areas with low POI density)
    - In high density areas, you might find too many points in the same cluster. 
    - in low density areas, it might give you better results

It may be possible to cluster different areas separately (e.g. inner city alone, and outer city alone) with different parameters
